File: 0704-binary-search/0704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        
        if(nums.length == 1){
            if(nums[0] == target){
                return 0;
            }else{return -1;}
        }
        
        int high = nums.length -1;
        int low = 0;
        
        while(low <= high){
            
            int mid = high + (low-high)/2;
            
            
            if(nums[mid] == target){
                return mid;

            }else if(nums[mid] > target){
                high = mid -1;
                
            }else if(nums[mid] < target){
            
               low = mid+1;
                
            }
            
        }
        return -1;
        
        
    }
    
  
}

File: 0704-binary-search/0704-binary-search.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        
        left = 0
        right = len(nums)-1
        if nums[left]== target:
            return left
        if nums[right] == target:
            return right
        while left < right:
            mid = (left+right)/2
            
            if(nums[mid] == target):
                return mid
            elif nums[left] == target:
                return left
            elif nums[right] == target:
                return right
            elif nums[mid] > target:
                right = mid -1
            else:
                left = mid+1
                
            
        
        
        3
        
        
        return -1
        